src/plugins/unified_histogram/public/services/lens_vis_service.ts

```diff
@@ -563,13 +609,15 @@ export class LensVisService {
       dataViewId: dataView.id,
       timeField: dataView.timeFieldName,
       timeInterval: isTextBased ? undefined : timeInterval,
-      breakdownField: isTextBased ? undefined : breakdownField?.name,
+      breakdownField: breakdownField?.name,
     };
 
+    const breakdownColumn = columns?.find((column) => column.name === breakdownField?.name);
+
     const currentQuery =
       suggestionType === UnifiedHistogramSuggestionType.histogramForESQL && isTextBased && timeRange
         ? {
-            esql: this.getESQLHistogramQuery({ dataView, query, timeRange }),
+            esql: this.getESQLHistogramQuery({ dataView, query, timeRange, breakdownColumn }),
```

Comment:
```suggestion
            esql: this.getESQLHistogramQuery({ dataView, query, timeRange, breakdownColumn:  breakdownField?.name ? columns?.find((column) => column.name === breakdownField.name) : undefined}),
```

Same, iterate only if necessary.

Reply:
Done https://github.com/elastic/kibana/pull/193820/commits/9e731dfc361a48659f74ee8b3896cdc0184cbf79